File: probes/probe.php
<?php

//** @see https://stackoverflow.com/a/43056278 */
ini_set('serialize_precision', -1);

class Probe
{
    /**@var string */
    protected
        $host,
        $env,
        $probe;

    public function __construct() {
        $this->host = getenv('PROBE_HOSTNAME');
        $this->env = getenv('PROBE_ENV');
        $this->probe = getenv('PROBE_NAME');
    }

    public function sendResults($results) {
        echo json_encode([
            'labels' => [
                'hostname' => $this->host,
                'env' => $this->env,
                'probe' => $this->probe
            ],
            'results' => $results
        ]);
    }
}
?>

File: probes/python/probe.py
# coding: utf-8

import fileinput
import os
import sys

try:
    import json
except ImportError:
    import simplejson as json 

class Probe:
    def __init__(self): # Notre méthode constructeur
        self.host = os.environ['PROBE_HOSTNAME']
        self.env = os.environ['PROBE_ENV']
        self.probe = os.environ['PROBE_NAME']
    
    def getConfig(self):
        return sys.stdin.read()
    
    def sendResults(self, results, additionalLabels = {}):
        labels = {'hostname': self.host, 'env': self.env, 'probe': self.probe, 'probe_args': ''}
        labels.update(additionalLabels)
        print(json.dumps({'labels': labels, 'results': results}))

